src/wp-includes/embed.php

```diff
@@ -411,10 +411,17 @@ function get_post_embed_url( $post = null ) {
 		return false;
 	}
 
-	$embed_url     = trailingslashit( get_permalink( $post ) ) . user_trailingslashit( 'embed' );
-	$path_conflict = get_page_by_path( str_replace( home_url(), '', $embed_url ), OBJECT, get_post_types( array( 'public' => true ) ) );
+	$is_published = 'publish' === $post->post_status;
 
-	if ( ! get_option( 'permalink_structure' ) || $path_conflict ) {
+	if ( $is_published ) {
```

Comment:
You never reuse `$is_published`, I'd use the actual condition right in line 416 and drop the assignment in 414.